hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_order_data_manager.py

```diff
@@ -0,0 +1,284 @@
+        self._order_status_running_tasks: List[Task] = []
+        self._order_status_polling_task: Optional[Task] = None
+
+    async def ready(self):
+        await self._ready.wait()
+
+    @property
+    def is_ready(self):
+        return self._is_ready
+
+    @property
+    def started(self):
+        return self._started
+
+    async def start(self):
+        if not self._started:
+            self._started = True
+            await self._initialize_order_books()
+            self._orderbook_polling_task = asyncio.create_task(self._poll_order_book_updates())
+
+            await self._initialize_order_status_streams()
+
+            await asyncio.sleep(self._start_timeout)
+
+            self._is_ready = True
+            self._ready.set()
+
+    async def stop(self):
+        await self._provider.close()
+        if self._orderbook_polling_task is not None:
+            self._orderbook_polling_task.cancel()
+            self._orderbook_polling_task = None
+        for task in self._order_status_running_tasks:
+            if task is not None:
+                task.cancel()
+        self._order_status_running_tasks.clear()
+
+    async def _initialize_order_books(self):
+        await self._provider.connect()
+        for trading_pair in self._trading_pairs:
+            initialized = False
+            for i in range(5):
+                blxr_orderbook = await self._provider.get_orderbook(market=trading_pair, limit=5, project=OPENBOOK_PROJECT)
+                if blxr_orderbook.market != "":
+                    self._apply_order_book_update(blxr_orderbook)
+                    initialized = True
+
+                    break
+            if not initialized:
+                raise Exception(f"orderbook for {trading_pair} not initialized")
+
+    async def _initialize_order_status_streams(self):
+        for trading_pair in self._trading_pairs:
+            self._markets_to_order_statuses.update({trading_pair: {}})
+
+            initialize_order_stream_task = asyncio.create_task(
+                self._poll_order_status_updates(trading_pair=trading_pair)
+            )
+            self._order_status_running_tasks.append(initialize_order_stream_task)
+
+    async def _poll_order_book_updates(self):
+        await self._provider.connect()
+        order_book_stream = self._provider.get_orderbooks_stream(
+            markets=self._trading_pairs, limit=5, project=OPENBOOK_PROJECT
+        )
+
+        async for order_book_update in order_book_stream:
+            self._apply_order_book_update(order_book_update.orderbook)
+
+    async def _poll_order_status_updates(self, trading_pair: str):
+        await self._provider.connect()
+        order_status_stream = self._provider.get_order_status_stream(
+            market=trading_pair, owner_address=self._owner_address, project=OPENBOOK_PROJECT
+        )
+
+        while True:
+            order_status_update = await order_status_stream.__anext__()
```

Comment:
usually you want to avoid calling `__anext__()` directly. use `async for`, since it handles stream closing behaviors properly

Reply:
When I was debugging this using a local Solana Trader API, it seemed that `async for` didn't start the order status stream function, but `__anext__()` did.

I think usually `async for` would work, but here we are running it inside a task so it may be different.

Reply:
it shouldn't be different. `async for` and `while + __anext__()` do the exact same thing; it's just much more idiomatic to do `async for`. Can you try it again here and see if it still doesn't work?